packages/paymasters/src/HashCashApproval.ts

```diff
@@ -0,0 +1,98 @@
+import { keccak256, toBN } from 'web3-utils'
+import { RelayRequest } from '@opengsn/common/dist/EIP712/RelayRequest'
+import abi from 'web3-eth-abi'
+import HashcashDifficulty from '@opengsn/paymasters/build/contracts/HashcashPaymaster.json'
+import IForwarder from '@opengsn/common/dist/interfaces/IForwarder.json'
+
+/**
+ * low-level hashcash calculation for the given address and nonce
+ * This value should be passed as approvalData for the HashcashPaymaster
+ * @param senderAddress the address of the sender
+ * @param senderNonce the current nonce of the sender
+ * @param difficulty target difficulty to meet
+ * @param interval call the callback every that many iterations
+ * @param callback async callback to call. return "false" to abort. true to continue
+ * @return the approvalData value (bytes32 hash, uint256 counter)
+ */
+export async function calculateHashcash (senderAddress: string, senderNonce: string, difficulty: any, interval?: number, callback?: any): Promise<string> {
+  const diffMax = toBN(1).shln(256 - difficulty)
+  let hashNonce = 0
+  let intervalCount = 0
+  while (true) {
+    // @ts-ignore
```

Comment:
why changed all ts-error to ignore ? ts-error **requires** an error (and thus should be removed when there's no more problem), while ts-ignore well, ignores everything.